addons/web/static/src/core/l10n/translation.js

```diff
@@ -39,6 +40,20 @@ export function _t(term, ...values) {
         if (values.length === 0) {
             return translation;
         }
+        // Localize list arguments
+        if (values.length === 1 && Object.prototype.toString.call(values[0]) === "[object Object]") {
+            Object.entries(values[0]).forEach(([key, value]) => {
+                if (Array.isArray(value)) {
+                    values[0][key] = formatList(value);
+                }
+            });
+        } else if (values.length > 0) {
+            values.forEach((value, index, arr) => {
+                if (Array.isArray(value)) {
+                    arr[index] = formatList(value);
+                }
+            });
+        }
```

Comment:
Wouldn't it be possible to check symbol iterator to accept any kind of iterator and not just arrays?